File: dcp-solutions/2024-11-10/solution.rb
```ruby
def myPow(x, y)
  if x == 0
    return 0
  end

  result = 1
  while y > 0 do
    if y % 2 == 1
      result *= x
    end
    x *= x
    y /= 2
  end

  result
end
```

File: dcp-solutions/2024-11-10/solution.js
```javascript
function myPow(x, y) {
    if (x == 0) return 0;
    let absY = Math.abs(y);
    let result = 1;
    
    while(absY > 0){
        if((absY&1) != 0){
            result *= x; 
        }
        x = x * x; 
        absY >>= 1; 
    }

    if (y < 0){
        return 1 / result;
    } else {
        return result;
    }
}
```